src/crafting_gui.cpp

```diff
@@ -557,7 +557,7 @@ const recipe *select_crafting_recipe( int &batch_size )
             }
             batch = !batch;
             if( batch ) {
-                batch_line = line;
+                batch_line = (batch_line >= (int)current.size() ? 0 : line);
```

Comment:
Build is failed due to astyle regression. You need to add whitespaces before '(' & ')' (except the case they are placed together, i.e. '()').